//
//  GoogleFindMyTools - A set of tools to interact with the Google Find My API
//  Copyright © 2024 Leon Böttger. All rights reserved.
//

syntax = "proto3";

message Time {
    uint32 seconds = 1;
    uint32 nanos = 2;
}

message LocationReport {
  SemanticLocation semanticLocation = 5;
  GeoLocation geoLocation = 10;
  Status status = 11;
}

message SemanticLocation {
  string locationName = 1;
}

enum Status {
  SEMANTIC = 0;
  LAST_KNOWN = 1;
  CROWDSOURCED = 2;
  AGGREGATED = 3;
}

message GeoLocation {
  EncryptedReport encryptedReport = 1;
  uint32 deviceTimeOffset = 2;
  float accuracy = 3;
}

message EncryptedReport {
    bytes publicKeyRandom = 1;
    bytes encryptedLocation = 2;
    bool isOwnReport = 3;
}

message GetEidInfoForE2eeDevicesRequest {
    int32 ownerKeyVersion = 1;
    bool hasOwnerKeyVersion = 2;
}